build(docker): add docker image creation configs

komi-eulisa opened the pull request:
- include: add a docker-compose file
- include: add instructions on how to run docker container

komi-eulisa committed:
build(docker): add docker image creation configs

- include: add a docker-compose file
- include: add instructions on how to run docker container

zugazagoitia-eulisa commented on Dockerfile:
```diff
@@ -0,0 +1,30 @@
+    && mkdir -p data temp ../logs config \
+    && chown -R ${USERNAME}:${USER_GROUP} ${APP_FOLDER} \
+    && chown -R ${USERNAME}:${USER_GROUP} /logs \
+    && rm config/connector.properties
```
Why is this needed?

komi-eulisa replied:
fixed

zugazagoitia-eulisa commented on README.md:
````diff
@@ -1 +1,173 @@
+Pull the image by running:
+
+```shell
+docker pull https://scm.ecodex.eu/connector:latest
````
Let's review these coordinates before merging

komi-eulisa replied:
Indeed

komi-eulisa replied:
I added this:
https://scm.ecodex.eu/docker/connector:latest

zugazagoitia-eulisa commented on README.md:
```diff
@@ -1 +1,173 @@
+
+### Already built image
+
+A built image of the connector is available here: [here](https://scm.ecodex.eu[]).
```
Check link before merging

komi-eulisa replied:
Indeed

komi-eulisa replied:
Added this: https://scm.ecodex.eu/docker/connector:latest

zugazagoitia-eulisa commented on README.md:
````diff
@@ -1 +1,173 @@
+Build the connector project in production mode by running:
+
+```shell
+mvn build install -Pproduction
````
`install`???

komi-eulisa replied:
Fixed

zugazagoitia-eulisa commented on README.md:
````diff
@@ -1 +1,173 @@
+Build the connector image by running:
+
+```shell
+docker build -t connector:ltest .
````
```suggestion
docker build -t connector:latest .
```

komi-eulisa replied:
Fixed

zugazagoitia-eulisa commented on README.md:
````diff
@@ -1 +1,173 @@
+```
+  /domibusConnectorDistribution/target/domibusConnector/documentation/database-scripts/initial/MySQL
+```
+* in the `sql` subfolder rename the files as follows:
````
Can we rename the sources of manually created SQL scripts to `0_` `1_` and `2_` ?

komi-eulisa replied:
is this OK ?
  * `mysql_initial.sql` -> `0_mysql_initial.sql`
  * `mysql_initial_data.sql` -> `1_mysql_initial_data.sql`
  * `mysql_quartz.sql` -> `2_mysql_quartz.sql`

komi-eulisa replied:
Fixs done by default in the source code. No need to rename the sql files

zugazagoitia-eulisa commented on docker-compose.yml:
```diff
@@ -0,0 +1,42 @@
+            - "3306"
+        volumes:
+            # based on your DBMS, initial scripts can be found in documentation/database-scripts/initial
+            # copy them and renamed them sequentially and placed them in sql folder
```
Reflect the change here if files are renamed in source

komi-eulisa replied:
Done

zugazagoitia-eulisa commented on docker-compose.yml:
```diff
@@ -0,0 +1,42 @@
+         #   context: ./
+         #   dockerfile: Dockerfile
+        container_name: connector
+        image: connector:latest
```
Maybe we should reference the JFROG image directly?

komi-eulisa replied:
For time being, I prefer doing it this way.
may be we can improve it in a near future

zugazagoitia-eulisa commented on domibusConnectorBuilder/domibusConnectorBuilderStandalone/src/main/docker/Dockerfile:
If we're not using it anymore, maybe we should remove it. The java version is also wrong

komi-eulisa replied:
Done

zugazagoitia-eulisa commented on pom.xml:
```diff
@@ -858,7 +859,18 @@
                 <!-- other -->
+                <plugin>
+                    <groupId>com.google.cloud.tools</groupId>
+                    <artifactId>jib-maven-plugin</artifactId>
```
Can we remove it?

komi-eulisa replied:
Done

komi-eulisa committed:
build(docker): improve the instruction

- include: improve the dockerisation readme instructions
- include: rename the sql files in the source code for better comprehension

komi-eulisa committed:
build(docker): add building and jfrog push instructions

zugazagoitia-eulisa commented on .github/workflows/docker.yaml:
```diff
@@ -0,0 +1,8 @@
+name: Docker Action
+on:
+  push:
```
Just on develop for now

zugazagoitia-eulisa replied:
Also, the workflow lives in `e-CODEX/workflows` now

komi-eulisa replied:
Done

zugazagoitia-eulisa reviewed (changes requested):
With the workflow change we should be ready to merge

komi-eulisa committed:
build(docker): build image only on devlop branch
